nflow-engine/src/main/java/io/nflow/engine/workflow/curated/MaintenanceWorkflow.java

```diff
@@ -0,0 +1,47 @@
+package io.nflow.engine.workflow.curated;
+
+import io.nflow.engine.service.MaintenanceConfiguration;
+import io.nflow.engine.service.MaintenanceResults;
+import io.nflow.engine.service.MaintenanceService;
+import io.nflow.engine.workflow.definition.NextAction;
+import io.nflow.engine.workflow.definition.StateExecution;
+import io.nflow.engine.workflow.definition.StateVar;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.schedule;
+import static io.nflow.engine.workflow.definition.NextAction.moveToState;
+
+/**
+ * Workflow that cleans up workflow instances periodically.
+ */
+@Component
+public class MaintenanceWorkflow extends CronWorkflow {
+  public static final String MAINTENANCE_WORKFLOW_TYPE = "nFlowMaintenance";
+  public static final String VAR_MAINTENANCE_CONFIGURATION = "config";
+
+  @Inject
+  MaintenanceService maintenanceService;
+
+  public MaintenanceWorkflow() {
+    super(MAINTENANCE_WORKFLOW_TYPE);
+    setDescription("Clean up workflow instances periodically.");
+  }
+
+  public NextAction doWork(@SuppressWarnings("unused") StateExecution execution, @StateVar(value = VAR_MAINTENANCE_CONFIGURATION, readOnly = true) MaintenanceConfiguration conf) {
+    MaintenanceResults results = maintenanceService.cleanupWorkflows(conf);
+    StringBuilder sb = new StringBuilder(64);
```

Comment:
This will look weird when there was nothing to do. Remove "Maintenance:" from the beginning?